classes/external/get_data_merge_requests.php

```diff
@@ -0,0 +1,150 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+namespace tool_mergeusers\external;
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once("{$CFG->libdir}/externallib.php");
+use \tool_mergeusers\merge_request;
+use external_function_parameters;
+use external_multiple_structure;
+use external_single_structure;
+use external_value;
+/**
+ * Version information
+ *
+ * @package     tool
+ * @subpackage  mergeusers
+ * @author      Nicola Vallinoto, Liguria Digitale
+ * @author      Jordi Pujol-Ahulló, SREd, Universitat Rovira i Virgili
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class get_data_merge_requests extends \external_api {
+    /**
+     * Webservice tool_mergeusers_get_data_merge_request parameters
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+        return new external_function_parameters([
+            'mergeusers' => new external_single_structure([
+                'removeuserfield' => new external_value(
+                    PARAM_TEXT, 'Remove user field', VALUE_OPTIONAL),
+                'removeuservalue' => new external_value(
+                    PARAM_RAW, 'Remove user value', VALUE_OPTIONAL),
+                'removeuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the remove user id', VALUE_OPTIONAL),
+                'keepuserfield' => new external_value(
+                    PARAM_TEXT, 'Keep user field', VALUE_OPTIONAL),
+                'keepuservalue' => new external_value(
+                    PARAM_RAW, 'Keep user value', VALUE_OPTIONAL),
+                'keepuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the keep user id', VALUE_OPTIONAL),
+                'id' => new external_value(
+                    PARAM_INT, 'Identifier of the merge request', VALUE_OPTIONAL),
+                'status' => new external_value(
+                    PARAM_INT, 'Status of the merge request', VALUE_OPTIONAL)
+            ])
+        ]);
+    }
+    /**
+     * Return data of the custom queue of the merge request
+     *
+     * @param $mergeusers id of the merge users request
+     *
+     * Return array with data of the custom queue of the merge request
+     * @param $id
```

Comment:
You are confusing the use of `@param` here, well used for the case of `@param $mergeusers`. Here you are describing  the content of every position of the returned array. Instead of just describing it, why not say that your are returning an array of merge_requests?

Internally, you would only need to wrap every stdClass instance returned by $DB->get_records() into an instance of merge_request.